include/tune_quda.h

```diff
@@ -150,21 +150,26 @@ namespace quda {
       }
     }
 
+    auto setSharedBytes(TuneParam &param) const
+    {
+      int nthreads = param.block.x * param.block.y * param.block.z;
+      param.shared_bytes = std::max(sharedBytesPerThread() * nthreads, sharedBytesPerBlock(param));
```

Comment:
@maddyscientist Thinking out loud, is the requirement that you set only one of `sharedBytesPerThread` and `sharedBytesPerThread` (which seems implicit in the use of `std::max` here) enforced anywhere?

If that requirement isn't imposed, should this be (and always have been) a sum, not a max?

Reply:
Funnily enough I was just thinking about this on my morning run.  We should probably either make it a sum or enforce than one of them is zero.  I don’t think the per-block shared bytes is used much (if anywhere at present).

Reply:
Per-block is used in `coarse_op.cuh` and `block_transpose.in.cu`, but I can't think of anywhere that _both_ per-block and per-thread is used.

That said, I think we should make it a sum as opposed to enforcing that one of them is zero. I can't think of any subtleties in letting them both be non-zero and I'd rather just have that "fix" in now as opposed to having to deal with it at some hypothetical future point.

Reply:
Resolved with `checkSharedBytes` function.  For now we just forbid both shared memory per thread and block, and we can reconsider this if necessary in the future.